### PR TITLE
#165305758 User profiles(Users)

### PR DESCRIPTION
**Pull request by a-braham**
#### Description
When a user registers into the platform, a profile is created with the default registration details(username). A user can update the rest of the details(first name, last name, image, bio) on his profile. A user can then get a profile for another user while logged in.

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [ ] End to end
- [x] Integration

#### Checklist:
- [ ] My code follows the style guide for this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
[#165305758](https://www.pivotaltracker.com/n/projects/2326460/stories/165305758)

### CODE REVIEW

File: .env-example [hunk 1]
```diff
@@ -4,3 +4,9 @@ DATABASE_NAME=your database name
 DATABASE_USER=your user name
 DATABASE_PASSWORD=your user password
 DATABASE_HOST=your host
+
+
+# Cloudnary environment configurations
+export CLOUDINARY_NAME=''
```
Comment (IssaIan) [inline .env-example:11]: Remember to add these environment variables on travis-ci.org and travis-ci.com to ensure your build passes.

File: authors/apps/profiles/tests/test_profiles.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        self.is_authenticated("adam@gmail.com", "@Us3r.com")
+        response = self.update_user_profile(
+            "Abraham", "Kamau", "I love history")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_profile_while_loggedout(self):
+        """
+        Test method for updating while user is not authenticated
+        """
+        response = self.update_user_profile(
```
Comment (IssaIan) [inline authors/apps/profiles/tests/test_profiles.py:42]: You could consider testing whether a user can update username. Also, you are only testing the status codes returned. Consider testing the messages returned too.

**Review by IssaIan — CHANGES_REQUESTED** [review]

File: authors/apps/profiles/tests/basetests.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        """
+        test method for all settings
+        """
+        self.user = User.objects.create_user(
+            username="adam",
+            email="adam@gmail.com",
+            password="@Us3r.com"
+        )
+
+    # def is_authenticated(self):
```
Comment (Ogutu-Brian) [inline authors/apps/profiles/tests/basetests.py]: Remove the commented code

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]

File: authors/apps/profiles/tests/test_profiles.py (hunk 2 above)
Comment (a-braham) [reply authors/apps/profiles/tests/test_profiles.py:42]: @IssaIan There is no username field in the database, the username is gotten from the User table which, to Profiles is a ForeignKey hence what is being saved in the Profile is just but the User Id

**Review by Ogutu-Brian — APPROVED** [review]
LGTM

**Review by Ogutu-Brian — COMMENTED** [review]
Fix the merge conflicts and the failing tests

**Review by Amoswachira — COMMENTED** [review]
@a-braham  you have failing tests, please fix them

**Review by Amoswachira — APPROVED** [review]
LGTM

**Review by IssaIan — APPROVED** [review]
LGTM

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]
Hello @a-braham ,
After you rebased this branch with develop, I realised that the endpoints you added are not documented on swagger, neither are they updated on the README file. Would you update this for future reference.

**Review by Ogutu-Brian — APPROVED** [review]
LGTM

**Review by Ogutu-Brian — COMMENTED** [review]
Update the README

**Review by matthenge — APPROVED** [review]
LGTM

**Comment by mdn-network** [comment]
Tested the updating and getting of a single profile everything works fine

**Pull request merged**